<p>###Homology Detection in Bioinformatics</p>

<p>A lot of my work is based on the tabular output of the tool BLAST (Basic Local Alignment Search Tool). This tool is used to search query sequences against the database of DNA/protein sequences and the output contains following columns</p>
<div class="highlighter-rouge"><pre class="highlight"><code>&lt;qseqid&gt; &lt;sseqid&gt; &lt;pident&gt; &lt;length&gt; &lt;mismatch&gt; &lt;gapopen&gt; &lt;qlen&gt; &lt;qstart&gt; &lt;qend&gt; &lt;slen&gt; &lt;sstart&gt; &lt;send&gt; &lt;evalue&gt; &lt;bitscore&gt;

</code></pre>
</div>
<p>The first 2 columns contain the IDs of query and subject sequence that is found as ‘hit’ in the database search with the query. The remaining cols contain various types of quantities that describe the quality of matching between the query and the subject sequence.
One of the main uses of the BLAST tabular output is to find and infer homologous sequences for each query
sequence. This can be achieved by setting user defined cutoffs on the numeric quantites in columns 4-12 where all the subject sequences (in column 2) from the database that satisfy the cutoffs will be regarded as true homologs of the query sequence in column 1</p>

<p>If I have to define a function for inferring homologuous sequences, literate programming can help me to describe and present the detail workings of the function, like the number and type of arguments and the return formats. The power of literate programming can also help me to explain, in a clear and natural way, the exact workflow of the function when a particular set of cutoff parameters are specified. For example, if the user provides pident (percent identity) cutoff value, all the query - subject pairs that are above this cutoff will be retained and for each query, the set of subject sequences will be returned as homologs of the query.</p>
